edweiss-app/__test__/app/(app)/lectures/remotecontrol/index.test.tsx

```diff
@@ -0,0 +1,232 @@
+
+// ------------------------------------------------------------
+// -----------------  Mocking dependencies    -----------------
+// ------------------------------------------------------------
+
+// `t` to return the key as the translation
+jest.mock('@/config/i18config', () => ({
+    __esModule: true,
+    default: jest.fn((key) => key),
+}));
+
+// Expo router
+jest.mock('expo-router', () => ({
+    router: { push: jest.fn() },
+    Stack: {
+        Screen: jest.fn(({ options }) => (
+            <>{options.title}</>
+        )),
+    },
+    useLocalSearchParams: jest.fn(() => ({ courseNameString: 'testCourse', lectureIdString: 'testLectureId' }))
+}));
+
+// Application navigation
+jest.mock('@react-navigation/native-stack', () => {
+    const actualNav = jest.requireActual('@react-navigation/native-stack');
+    return {
+        ...actualNav,
+        createNativeStackNavigator: jest.fn(() => ({
+            Navigator: ({ children }: { children: ReactNode; }) => <>{children}</>,
+            Screen: ({ children }: { children: ReactNode; }) => <>{children}</>,
+        })),
+    };
+});
+
+// Authentication mock
+jest.mock('@/contexts/auth', () => ({
+    useAuth: jest.fn(),
+}));
+
+// Library Settings mock 
+jest.mock('react-native/Libraries/Settings/Settings', () => ({
+    get: jest.fn(),
+    set: jest.fn(),
+}));
+
+// Async storage for the pdf as it is stored in FBcloud storage
+jest.mock('@react-native-async-storage/async-storage', () => ({
+    setItem: jest.fn(),
+    getItem: jest.fn(),
+    removeItem: jest.fn(),
+}));
+
+// Firebase mock
+jest.mock('@/config/firebase', () => ({
+    CollectionOf: jest.fn(),
+    getDownloadURL: jest.fn(),
+    callFunction: jest.fn()
+}));
+
+// react-native-blob-util to prevent NativeEventEmitter errors
+jest.mock('react-native-blob-util', () => ({
+    fs: {
+        dirs: {
+            DocumentDir: 'mockDocumentDir',
+        },
+    },
+    config: jest.fn(() => ({
+        fetch: jest.fn(),
+    })),
+}));
+
+// Firebase Messaging to avoid NativeEventEmitter errors
+jest.mock('@react-native-firebase/messaging', () => ({
+    __esModule: true,
+    default: jest.fn(() => ({
+        onMessage: jest.fn(),
+        onNotificationOpenedApp: jest.fn(),
+        getInitialNotification: jest.fn(),
+    })),
+}));
+
+// Updated mock for expo-screen-orientation
+jest.mock('expo-screen-orientation', () => ({
+    lockAsync: jest.fn(),
+    unlockAsync: jest.fn(),
+    OrientationLock: { LANDSCAPE: 'LANDSCAPE', PORTRAIT_UP: 'PORTRAIT_UP' },
+    addOrientationChangeListener: jest.fn((callback) => {
+        callback({ orientationInfo: 'mockOrientation' });
+        return {
+            remove: jest.fn(),
+        };
+    }),
+    removeOrientationChangeListener: jest.fn(),
+}));
+
+// Voice handler
+jest.mock('@react-native-voice/voice', () => ({
+    start: jest.fn(),
+    stop: jest.fn(),
+    onSpeechResults: jest.fn(),
+    onSpeechError: jest.fn(),
+}));
+
+
+// 
```

Comment:
Minor detail due to consistency, just something that seems to be missing, since you always make such great documentations this could be interesting to consider to maintain it clean

Reply:
Yes, you are right. Moreover, this mock is very important as it provide the `usePrefetchedDynamicDoc` for the tests 🧪. It uses a double array to match the `lectureDoc`. I will add a comment, so that in the future we can keep trace of your remark.